tests/run_tests.sh

```diff
@@ -0,0 +1,351 @@
+#!/bin/bash
+
+#USAGE: ./run_tests.sh [0|1|2] [last_test_number]
+#First argument is a debug value. 1 adds --debug to htseq2multx.pl commands to
+#cause the script to echo the fastq-multx command.  2 also causes the script to
+#stop upon the end of the first failed test.  3 echoes all the test script's
+#commands
+
+TEST_DATA=test_data
+TEST_OUTPUT=test_output
+BARCODE_SPLITTER=../htseq2multx.pl
+DEBUG=0
+DEBUG_OPT=""
+if [ $# -ne 0 ] && [ $1 -ne 0 ]; then
+  DEBUG=$1
+  DEBUG_OPT="--debug"
+fi
+LAST_TEST=""
+if [ $# -gt 1 ]; then
+  LAST_TEST="test_$2"
+fi
+rm -rf "${TEST_OUTPUT}"
+mkdir -p "${TEST_OUTPUT}"
+
+#Perl code to allow fastq-multx debug message to not cause a "difference"
+PERL_DEBUG_FILTER='$c=0;$d="";while(<STDIN>){if($_ !~ /#.*fastq-multx|^1d0$/){print;$c++}elsif(/#.*fastq-multx/){$d.=$_}}if($c && $ARGV[0]){$d =~ s/< #/#/g;print "\033[1;94m$d\033[0m"}exit($c)'
```

Comment:
Perhaps it would be better to just change the test output file? I'm not sure there is value in keeping the test output files the same when the output is intentionally different.

Reply:
Of course, right after I post the review, I sorted out the issue. `cat` doesn't handle zip files on Ubuntu (not sure it does anywhere). `zcat` isn't always available, at least not as often as `gunzip` is.

Reply:
I'm not sure the intent here is properly conveyed.  (Or perhaps your second comment here is in the wrong thread?)  This is only for debug mode. I added the ability for the perl script to print the fastq-multx call with a prepended "#" comment character because I wanted to see the call for tests that were failing.  This perl script one-liner is only useful for when `DEBUG` on line 12 is not `0`, which is controlled by the first script argument.  If you wish, I could remove the DEBUG-related stuff entirely.  But I find it handy for figuring out why a test failed.

Reply:
Sorry, guess I missed the debit stuff. That's fine to leave then.

As for the zcat in the test, I realized that the test was intended to test getting gzipped file on input. `zcat` is just `gunzip -c`, so it's unzipping the file. That's not really the intent of the test, but perhaps it just won't work since it relies on filenames to detect gzip input.

Reply:
Oh, so you're suggesting that cat relies on the presence of the extension in the same way that fastq-multx does?  That could be.  But even so, that would also apply to barcode_splitter.  Can you successfully run this corresponding test for barcode_splitter on ubuntu?  If the issue on my mac is the same as on ubuntu, then I assume you cannot, since it was `cat` that gave me the error I encountered, which prompted me to use the if conditional to set zcat.  We're talking about test 25, right?

Reply:
No, `cat` doesn't care and works just fine on text or binary files. The original test seems to have been written to test barcode splitter's ability to handle gzipped input from a pipe. Changing it to use `zcar` or `gunzip -c` defeats that purpose.

Reply:
Oh!  I see!  Right. That didn't occur to me.  So... but like I said, on my mac, cat complains about the compressed file.  Let me rerun and get the error and figure out how to address it correctly.

Reply:
So there's no error from cat.  I wasn't remembering correctly.  There's just no gz data output.  I suspect there's more going on here.  I'll dig into it tomorrow.

Reply:
Figured it out.  The sub making the temporary file copies wasn't even being called because my conditional was using the opposite logic it should have had - plus, I was deleting the fd file name before the call.  Fix impending...